src/array.jl

```diff
@@ -211,7 +211,7 @@ end
                 r
             end for a in A]
             $A(DefaultRefType[refs...;],
-               CategoricalPool(newlevels, ordered && all(isordered, A)))
+               CategoricalPool(newlevels, ordered && all(a->isordered(a) || 0==length(levels(a)), A)))
```

Comment:
`isempty(levels(a)) || ...` should be slightly more efficient (and more concise).